isabelle/generated/SemanticsGuarantees.hs

```diff
@@ -8,33 +8,28 @@ import Prelude ((==), (/=), (<), (<=), (>=), (>), (+), (-), (*), (/), (**),
   zip, null, takeWhile, dropWhile, all, any, Integer, negate, abs, divMod,
   String, Bool(True, False), Maybe(Nothing, Just));
 import qualified Prelude;
-import qualified ByteString;
+import qualified Product_Type;
 import qualified MList;
 import qualified Arith;
 import qualified Product_Lexorder;
 import qualified SemanticsTypes;
 import qualified Orderings;
 
 less_eq_Party :: SemanticsTypes.Party -> SemanticsTypes.Party -> Bool;
-less_eq_Party (SemanticsTypes.Address uu) (SemanticsTypes.Role uv) = True;
-less_eq_Party (SemanticsTypes.Role uw) (SemanticsTypes.Address ux) = False;
-less_eq_Party (SemanticsTypes.Address x) (SemanticsTypes.Address y) =
-  ByteString.less_eq_ByteString x y;
-less_eq_Party (SemanticsTypes.Role x) (SemanticsTypes.Role y) =
-  ByteString.less_eq_ByteString x y;
-
-less_eq_Partya :: SemanticsTypes.Party -> SemanticsTypes.Party -> Bool;
-less_eq_Partya a b = less_eq_Party a b;
+less_eq_Party a b =
+  (case (a, b) of {
+    (SemanticsTypes.Address x, SemanticsTypes.Address aa) -> x <= aa;
+    (SemanticsTypes.Address _, SemanticsTypes.Role _) -> True;
+    (SemanticsTypes.Role _, SemanticsTypes.Address _) -> False;
+    (SemanticsTypes.Role x, SemanticsTypes.Role aa) -> x <= aa;
+  });
```

Comment:
I think pattern-matching here is much better than a case (even if it is more characters), but the other changes are good.

Justification:
- When you use <= later and simplify it, the simplifier will always inline this (because there are no demands on the parameters), so you will just get a blob every time, when the desirable/readable thing would be to either not simplify it at all, or simplify it if you can pattern match (you get much smaller expressions in both cases), and that is the result you would get if you used the pattern matching
- If you use pattern matching the lemmas generated will be one per clause, whereas here you get a single lemma for the whole thing
- Variables `a` and `b` are unnecessary, you could call `aa`, just `a`, if you hadn't used `a` identifier already

Reply:
solved